qiskit/circuit/controlledgate.py

```diff
@@ -182,3 +182,19 @@ def __eq__(self, other) -> bool:
     def inverse(self) -> 'ControlledGate':
         """Invert this gate by calling inverse on the base gate."""
         return self.base_gate.inverse().control(self.num_ctrl_qubits)
+
+    def control(self, num_ctrl_qubits=1, label=None, ctrl_state=None):
```

Comment:
I think it might be better to leave this functionality in add_control.py so that it's as consolidated as possible. If it's done there you may be able to avoid setting label in all the standard gates at least for the super() call.

Reply:
Something like this? 0cfb5cbf9